modules/stochastic_tools/src/reporters/AdaptiveMonteCarloDecision.C

```diff
@@ -90,25 +102,44 @@ AdaptiveMonteCarloDecision::execute()
   if (_ais)
   {
     const Real tmp = _ais->getUseAbsoluteValue() ? std::abs(_output_value[0]) : _output_value[0];
-    const bool output_limit_reached = tmp >= _output_limit;
+    bool output_limit_reached;
+
+    /* Checking whether a GP surrogate is used. If it is used, importance sampling is not performed
+    during the training phase of the GP and all proposed samples are accepted until the training
+    phase is completed. Once the training is completed, the importance sampling starts.
+    If a GP surrogate is not used, the standard proposal and acceptance/rejection is performed as
+    part of the importance sampling. */
+    bool restart_gp = 0;
+    output_limit_reached = tmp >= _output_limit;
+    if (_gp_used)
+    {
+      const int training_samples =
+          getUserObject<ActiveLearningGPDecision>("gp_decision").getTrainingSamples();
```

Comment:
I recommend making `training_samples` a member variable:
```cpp
const int * const _gp_training_samples;
```
Then in the constructor initialization:
```cpp
_gp_training_samples(_gp_used ? &getUserObject<ActiveLearningGPDecision>("gp_decision").getTrainingSamples() : nullptr)
```